lib/galaxy/tools/actions/__init__.py

```diff
@@ -236,7 +236,7 @@ def execute(self, tool, trans, incoming={}, return_job=False, set_output_hid=Tru
                 incoming[ "%s|__identifier__" % name ] = identifier
 
             for tag in data.tags:
-                if tag.user_tname == 'name':
+                if tag.user_tname == 'name' and tag.value not in [t.value for t in preserved_tags]:
```

Comment:
Hrmm.  This should be the same (now filtered) list passed forward to the output_collections handling (https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/actions/__init__.py#L409)

Got a minimal example with out-of-the-box tools I can try to see what you're seeing?